src/pe_reports/data/db_query.py

```diff
@@ -0,0 +1,202 @@
+    logging.error(f"psycopg2 traceback: {traceback} -- type: {err_type}")
+    logging.error(f"\nextensions.Diagnostics: {err}")
+
+    logging.error(f"pgerror: {err}")
+
+    logging.error(f"pgcode: {err}\n")
+
+
+def connect():
+    """Connect to PostgreSQL database."""
+    conn = None
+    try:
+        logging.info("Connecting to the PostgreSQL......")
+
+        conn = psycopg2.connect(**CONN_PARAMS_DIC)
+        logging.info("Connection successful................\n")
+
+    except OperationalError as err:
+        show_psycopg2_exception(err)
+        conn = None
+    return conn
+
+
+def close(conn):
+    """Close connection to PostgreSQL."""
+    conn.close()
+    return
+
+
+def get_orgs(conn):
+    """Query organizations table."""
+    try:
+        cur = conn.cursor()
+        sql = """SELECT * FROM organizations"""
+        cur.execute(sql)
+        pe_orgs = cur.fetchall()
+        cur.close()
+        return pe_orgs
+    except (Exception, psycopg2.DatabaseError) as error:
+        logging.error(f"There was a problem with your database query {error}")
+    finally:
+        if conn is not None:
+            close(conn)
+
+
+def query_hibp_view(conn, org_uid, start_date, end_date):
+    """Query 'Have I Been Pwned?' table."""
+    try:
+        sql = """SELECT * FROM vw_breach_complete
+        WHERE organizations_uid = %(org_uid)s
+        AND modified_date BETWEEN %(start_date)s AND %(end_date)s"""
+        df = pd.read_sql(
+            sql,
+            conn,
+            params={"org_uid": org_uid, "start_date": start_date, "end_date": end_date},
+        )
+        return df
+    except (Exception, psycopg2.DatabaseError) as error:
+        logging.error(f"There was a problem with your database query {error}")
+    finally:
+        if conn is not None:
+            close(conn)
+
+
+def query_domMasq(conn, org_uid, start_date, end_date):
+    """Query domain masquerading table."""
+    try:
+        sql = """SELECT * FROM dnstwist_domain_masq
+        WHERE organizations_uid = %(org_uid)s
+        AND date_observed BETWEEN %(start_date)s AND %(end_date)s"""
+        df = pd.read_sql(
+            sql,
+            conn,
+            params={
+                "org_uid": org_uid,
+                "start_date": start_date,
+                "end_date": end_date,
+            },
+        )
+        return df
+    except (Exception, psycopg2.DatabaseError) as error:
+        logging.error(f"There was a problem with your database query {error}")
+    finally:
+        if conn is not None:
+            close(conn)
+
+
+"""
+The 'table' parameter is used in query_shodan, query_darkweb and
+query_darkweb_cves functions to call specific tables that relate to the function name.
+The result of this implementation reduces the code base, the code reduction leads
+to an increase in efficiency by reusing the function by passing only a parameter to get
+the required information from the database.
+"""
+
```

Comment:
Sorry for the nitpick, but we have standardized on this comment style:
```suggestion
# The 'table' parameter is used in query_shodan, query_darkweb and
# query_darkweb_cves functions to call specific tables that relate to the
# function name.  The result of this implementation reduces the code base,
# the code reduction leads to an increase in efficiency by reusing the
# function by passing only a parameter to get the required information from
# the database.
```

I don't believe we have documented that stylistic choice anywhere, so apologies for springing it on you here.

Reply:
@dav,
Is the style based on line length? If so I can get that tidbit of information put in to our running list of items we all need to watch for. Thanks

Reply:
@dav3r ^^ meant for you

Reply:
Our pre-commit linters should be complaining about any lines longer than 88 characters (I have my local editor set to wrap lines at 80 characters), but the main thing I was talking about here was comments with `#` vs. block comments between `'''` lines.